File: app_mobile_legacy/src/main/java/com/kee0kai/thekey/domain/room/dao/SettingDao.java
package com.kee0kai.thekey.domain.room.dao;

import static androidx.room.OnConflictStrategy.REPLACE;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.kee0kai.thekey.domain.room.model.SettingPairEntry;

@Dao
public interface SettingDao {

    String TABLE_NAME = "settting_pairs";

    @Insert(onConflict = REPLACE)
    void update(SettingPairEntry entry);

    @Query("SELECT * FROM " + SettingDao.TABLE_NAME + " WHERE id = :id")
    SettingPairEntry get(long id);

}

File: core/src/main/kotlin/com/github/klee0kai/thekey/core/data/room/dao/SettingDao.kt
package com.github.klee0kai.thekey.core.data.room.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.github.klee0kai.thekey.core.data.room.entry.SettingPairEntry

@Dao
interface SettingDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun update(entry: SettingPairEntry)

    @Query("SELECT * FROM $TABLE_NAME WHERE id = :id")
    operator fun get(id: Int): SettingPairEntry?

    companion object {
        const val TABLE_NAME = "settting_pairs"
    }
}